src/riscv-integration.adoc

```diff
@@ -791,6 +805,23 @@ CHERI violations have the following order in priority:
 . Invalid address violation
 . Bounds violation (_Lowest_)
 
+[#mtval2,reftext="mtval2"]
+==== Machine Trap Value Register 2 (mtval2)
+
+The <<mtval2>> register is an MXLEN-bit read-write register, which is added as part of the
+Hypervisor extension. {cheri_base_ext_name} also requires the implementation of this CSR.
+
+When a CHERI fault is taken on any data memory access into into M-mode, <<mtval2>> is written
+with the MXLEN-bit address which caused the fault. This can be used together with the
```

Comment:
Why address is that? For example, if I load 4 bytes from address 0x10, but only bytes at 0x10 and 0x11 are in bounds while 0x12 and 0x13 are out-of-bounds, then what address do I get in mtval2? Is it 0x10 or 0x12 (or something else!)?

Reply:
the load/store address, which would normally be in mtval. the lowest byte of the access

Reply:
fixed - it follows the standard Xtval rules